### ISSUE TITLE
Smart Lighting motion sensors not working

### ISSUE THREAD

**Issue by bolencki13** [tsightler/ring-mqtt#84]
[Stand-alone ring motion sensors](https://shop.ring.com/products/smart-lighting-motion-sensor) or motion sensors part of the ring lights that are a part of the ring light suit are not triggering.

Expected functionality:
Receive an update to the state of the device in home assistant.

Perceived functionality:
Ring motion sensor appears to registers two binary sensors in home assistant for a single motion sensor. Neither appear to update their states when motion is found on the device. Ring's mobile application shows a history of motion yet home assistant does not.

**Comment by tsightler** [comment]
I don't actually own any Ring smart lighting and have no plans to purchase any at this time so the code for this was developed "blind".  Feedback from other users have indicated that it worked fine but that's all I have to go on.  Is this a new issue or something that previously worked for you?

If you are willing to provide debugging information I'm happy to try to dig into it, but I would most likely need debug output for at least these devices.  Also, do you see any debug output when motion is triggered on these devices.

I can't see any reason that the same device would be registered twice, but every light also has a motion sensor, is it possible that's why you see two sensors registered?

**Comment by tsightler** [comment]
Did a quick review of the motion status code (it's basically three lines) and did indeed find a typo on the property name for the motion status, which would likely explain why it wasn't working.  If you reinstall with the latest version (just hit rebuild if using the HASS.io addon) it might work now.  Please let me know.

Would still be interested in understanding the "two sensors" issue as well, I have no idea how that would be possible with the current code unless Ring somehow returns two separate devices.

**Comment by bolencki13** [comment]
No worries, man. I appreciate the effort. I was unable to get this functionality working on any version of ring-mqtt. I did try out the latest fix and also found some more details that may be helpful.

Sensor is showing in home assistant under two different identifiers: 
<img width="863" alt="Screen Shot 2020-08-07 at 11 06 35 AM" src="https://user-images.githubusercontent.com/9823280/89659686-355de600-d89e-11ea-9084-cd1ac96096be.png">

After looking at the logs for ring-mqtt and triggering motion manually, it appears the to be capturing the movement but possibly classifying it incorrectly. My best guess is the [conditional](https://github.com/tsightler/ring-mqtt/commit/4ed3c03ec15b1bea391ace8140854abbf7420ce1#diff-f1b898715b4cbc839f244319f2835636R90) may not be classified as "faulted". The reason I made this assumption is because the logs indicate motion is triggered by updating however it is always a state of OFF.

I'm happy to assist in any way possible. If you'd like logs or other device specific info, I would gladly share it with you.

**Comment by tsightler** [comment]
Since I didn't have any Ring smart lighting to test with I followed the logic in the [Homebridge plugin](https://github.com/dgreif/ring/blob/ab8764acdbaa710296ccd9de33667f98e7b10577/homebridge/beam.ts#L51) regarding the conditional.  It looks to me like this code is clearly using "faulted"

The easiest thing to do is probably to just add the line "debug(this.device.data)" to get the raw data output when motion is triggered.  It should become really obvious what value in the data is changing when motion is detected.

Right now I can't even fathom how it's creating two devices though.

**Comment by tsightler** [comment]
Looking at the screenshot, based on the fact that "Driveway" has no device attributes, that would indicate it is a light group.  Can you verify if that's the case?

**Comment by bolencki13** [comment]
That would make sense. The top one has no lights in it but it is a group named Driveway and only contains the motion sensor. I'll investigate more as to the constant OFF state.

**Comment by bolencki13** [comment]
Maybe I am misunderstanding the implementation... It looks like [you referenced a conditional](https://github.com/dgreif/ring/blob/ab8764acdbaa710296ccd9de33667f98e7b10577/homebridge/beam.ts#L51) for the motionStatus in the Homebridge plugin. In the [motion-sensor device](https://github.com/tsightler/ring-mqtt/blob/master/devices/motion-sensor.js#L45), I believe it checks for a property of faulted vs the conditional in home bridge. Am I reading that right?

**Comment by tsightler** [comment]
The motion-sensor.js code is for alarm motion sensors, i.e. devices of class sensor.motion.  The Smart Lighting devices are beam devices, i.e. devices of class motion-sensor.beam.  All code for beam devices, including motion sensors, is in beam.js, so you should only compare beam.js in ring-mqtt to beam.js in homebridge.

However, you can also look at properties for RingDeviceData in the API:

https://github.com/dgreif/ring/blob/ab8764acdbaa710296ccd9de33667f98e7b10577/api/ring-types.ts#L167

You will see that there is a property:
faulted?: boolean

This is used my sensor.motion devices and why the code can simply check for boolean state.  Further down you will see:

motionStatus?: 'clear' | 'faulted'

This is the property used by motion-sensor.beam devices (or at least it should be) and it's a string value so I'm using a conditional to compare the value of this property to "faulted" and setting state based on that.  At least, that's what the code is attempting to do and what the homebridge code also appears to do.

If you just add the debug line I mentioned above right above the `const motionState = this.device.data.motionStatus === 'faulted' ? 'ON' : 'OFF'` it should tell us pretty quickly what is happening.

**Comment by bolencki13** [comment]
Just ran this locally on my laptop (no docker), and it works!! Alexa started yelling at me that someone was coming down my driveway lol. Good to know my automation is working😁. I am going to start debugging Home assistant to see why the docker container isn't working...

**Comment by tsightler** [comment]
If you're doing a Docker pull from the repo its mostly likely because I haven't pushed the very latest changes there yet (Docker image is still 3.1.0).  I still need to make some fixes for the Docker support which I hope to do this weekend and push a new Docker image.

**Comment by bolencki13** [comment]
I've been using the [ring-mqtt-hassio-addon](https://github.com/tsightler/ring-mqtt-hassio-addon) you made as well. When I was rebuilding the image, it seems to be using a cache or something. The project instance of ring-mqtt within the running docker container wasn't updating to the latest master branch.

I followed these [instructions](https://gist.github.com/glamp/74188691c91d52770807) on modifying a docker image. I connected to the running container and manually pulled the master branch. By committing the changes and overwriting my local image of ring-mqtt-hassio-addon it seems to be running great. 

That was a great fix, thanks for the help!
